ash-window/src/lib.rs

```diff
@@ -118,6 +118,45 @@ pub unsafe fn create_surface(
     }
 }
 
+#[cfg(target_os = "windows")]
+/// Extensions necessary for creating a surface on the target platform.
+pub const TARGET_EXTENSIONS: [*const c_char; 2] = [
+    khr::Surface::name().as_ptr(),
+    khr::Win32Surface::name().as_ptr(),
+];
+#[cfg(any(
+    target_os = "linux",
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd"
+))]
+/// Extensions necessary for creating a surface on the target platform.
+pub const TARGET_EXTENSIONS: [*const c_char; 4] = [
+    khr::Surface::name().as_ptr(),
+    khr::WaylandSurface::name().as_ptr(),
+    khr::XlibSurface::name().as_ptr(),
+    khr::XcbSurface::name().as_ptr(),
+];
+#[cfg(any(target_os = "android"))]
+/// Extensions necessary for creating a surface on the target platform.
+pub const TARGET_EXTENSIONS: [*const c_char; 2] = [
+    khr::Surface::name().as_ptr(),
+    khr::AndroidSurface::name().as_ptr(),
+];
+#[cfg(any(target_os = "macos"))]
+/// Extensions necessary for creating a surface on the target platform.
+pub const TARGET_EXTENSIONS: [*const c_char; 2] = [
+    khr::Surface::name().as_ptr(),
+    ext::MetalSurface::name().as_ptr(),
+];
+#[cfg(any(target_os = "ios"))]
+/// Extensions necessary for creating a surface on the target platform.
+pub const TARGET_EXTENSIONS: [*const c_char; 2] = [
+    khr::Surface::name().as_ptr(),
+    ext::MetalSurface::name().as_ptr(),
+];
+
```

Comment:
Let's make these slices rather than arrays so it's harder to inadvertently statically depend on the length.

Reply:
Alright.